File: apps/ios-app/App/Core/FireBrowserTransportPrompt.swift
import Foundation

enum FireBrowserTransportPrompt {
    static func shouldPresent(
        kind: AuthRuntimeSignalKindState?,
        alreadyLatched: Bool
    ) -> Bool {
        kind == .askEnableBrowserTransport && !alreadyLatched
    }

    static func nextLatch(kind: AuthRuntimeSignalKindState?) -> Bool {
        kind == .askEnableBrowserTransport
    }
}

File: apps/android-app/src/main/java/com/fire/app/session/FireBrowserTransportPrompt.kt
package com.fire.app.session

import uniffi.fire_uniffi_session.AuthRuntimeSignalKindState
import uniffi.fire_uniffi_session.SessionState

object FireBrowserTransportPrompt {
    fun shouldPresent(kind: AuthRuntimeSignalKindState?, alreadyLatched: Boolean): Boolean {
        return kind == AuthRuntimeSignalKindState.ASK_ENABLE_BROWSER_TRANSPORT && !alreadyLatched
    }

    fun nextLatch(kind: AuthRuntimeSignalKindState?): Boolean {
        return kind == AuthRuntimeSignalKindState.ASK_ENABLE_BROWSER_TRANSPORT
    }

    fun signalKind(snapshot: SessionState): AuthRuntimeSignalKindState? {
        return snapshot.lastAuthRuntimeSignal?.kind
    }
}
